kani-compiler/src/kani_middle/transform/check_aliasing/body_mutation.rs

```diff
@@ -0,0 +1,28 @@
+use super::{Local, InstrumentationData, MirError, Body};
+
+pub struct BodyMutationPassState<'tcx, 'cache> {
+    values: Vec<Local>,
+    instrumentation_data: InstrumentationData<'tcx, 'cache>,
+}
+
+impl<'tcx, 'cache> BodyMutationPassState<'tcx, 'cache> {
+    pub fn new(values: Vec<Local>, instrumentation_data: InstrumentationData<'tcx, 'cache>) -> Self {
+        BodyMutationPassState { values, instrumentation_data }
+    }
+
+    pub fn instrument_locals(&mut self) -> Result<(), MirError> {
+        self.instrumentation_data.instrument_locals(&self.values)
+    }
+
+    pub fn instrument_instructions(&mut self) -> Result<(), MirError> {
+        self.instrumentation_data.instrument_instructions()?;
+        Ok(())
+    }
+
+    pub fn finalize(mut self) -> Body {
```

Comment:
This is the only method that adds extra functionality. I would rather have it as a single helper function instead of creating a separate struct for this.